setup.py

```diff
@@ -166,98 +166,98 @@ def run(self):
     setup_requires = []
     default_backend = None
 
-    # Go through all of the packages and figure out which ones we are
-    # going to build/install.
-    print_line()
-    print_raw("Edit setup.cfg to change the build options")
-
-    required_failed = []
-    good_packages = []
-    for package in mpl_packages:
-        if isinstance(package, str):
-            print_raw('')
-            print_raw(package.upper())
-        else:
-            try:
-                result = package.check()
-                if result is not None:
-                    message = 'yes [%s]' % result
-                    print_status(package.name, message)
-            except setupext.CheckFailed as e:
-                msg = str(e).strip()
-                if len(msg):
-                    print_status(package.name, 'no  [%s]' % msg)
-                else:
-                    print_status(package.name, 'no')
-                if not package.optional:
-                    required_failed.append(package)
-            else:
-                good_packages.append(package)
-                if (isinstance(package, setupext.OptionalBackendPackage) and
-                        package.runtime_check() and
-                        default_backend is None):
-                    default_backend = package.name
-    print_raw('')
-
-    # Abort if any of the required packages can not be built.
-    if required_failed:
+    # If the user just queries for information, don't bother figuring out which
+    # packages to build or install.
+    if (any('--' + opt in sys.argv for opt in
+        Distribution.display_option_names + ['help']) or
```

Comment:
This line should be indented one more so as to not coincide with the code indent in the block. But even so, it should be indented anyway because it's within the `any(`.